metals/src/main/scala/scala/meta/internal/pantsbuild/BloopPants.scala

```diff
@@ -354,156 +336,72 @@ private class BloopPants(
     } yield library
-    val libraryDependencyConfigs: List[String] =
-      List("test", "default", "shaded", "linux-x86_64", "thrift9")
-    val libraryDependencyClasspaths =
-      getLibraryDependencies(libraryDependencies, libraryDependencyConfigs)
-    val sourcesConfigs = List("sources")
-    val libraryDependencySources =
-      getLibraryDependencies(libraryDependencies, sourcesConfigs)
-
-    // Warn about unfamiliar configurations.
-    val knownConfigs = sourcesConfigs ::: libraryDependencyConfigs
-    val unknownConfigs = libraryDependencies
-      .flatMap(lib => lib.obj.keys.toSet -- knownConfigs)
-      .distinct
-    if (unknownConfigs.nonEmpty) {
-      val kinds = unknownConfigs.mkString(",")
-      println(
-        s"[warning] Unknown configs for target '${id}' with type '${target.targetType}': ${kinds}"
-      )
-    }
-
-    val out = bloopDir.resolve(BloopPants.makeFilename(id))
-    val classDirectory = out.resolve("classes")
-    val javaHome = Option(System.getProperty("java.home")).map(Paths.get(_))
-
-    val resources =
-      if (target.isResourceTarget) {
-        // Resources are relativized by the directory where the BUILD file exists.
-        Some(List(baseDirectory))
-      } else {
-        None
-      }
-
-    val test = if (target.pantsTargetType.contains("test")) {
-      val scalatest = C.TestFramework(
-        List(
-          "org.scalatest.tools.Framework",
-          "org.scalatest.tools.ScalaTestFramework"
-        )
-      )
-      // These test frameworks are the default output from running `show
-      // testFrameworks` in sbt (excluding spec2). The output from `./pants
-      // export` doesn't include the configured test frameworks.
-      val defaultTestFrameworks = List(
-        scalatest,
-        C.TestFramework(List("org.scalacheck.ScalaCheckFramework")),
-        C.TestFramework(List("com.novocode.junit.JUnitFramework"))
-      )
-      Some(
-        C.Test(
-          frameworks = defaultTestFrameworks,
-          options = C.TestOptions(
-            excludes = Nil,
-            arguments = List(
-              C.TestArgument(
-                List("-o"),
-                Some(scalatest)
-              )
-            )
-          )
-        )
-      )
-    } else {
-      None
-    }
 
-    val resolution: Option[C.Resolution] =
-      if (id.isBinaryDependency) {
-        binaryDependencySources ++= libraryDependencySources
-        binaryDependencySources ++= enclosingSourceDirectory(baseDirectory)
-        None
-      } else {
-        Some(
-          C.Resolution(
-            libraryDependencySources.iterator.map(newSourceModule).toList
-          )
-        )
-      }
+    val classpath = new mutable.LinkedHashSet[Path]()
+    classpath ++= (for {
+      dependency <- transitiveDependencies
+      if dependency.isTargetRoot
+    } yield dependency.classesDir(bloopDir))
+    classpath ++= (for {
+      dependency <- transitiveDependencies
+      if !dependency.isTargetRoot
```

Comment:
nit: I think indenting the `if` would make these filters more readable.

Reply:
This is how scalafmt formats this. Originally it used to indent `if` filters in for comprehensions but it got changed because it got weird in places like
```scala
for {
  a <- list()
  b <- list()
  if a > 2
  if b < 1
```